src/circuits/etable_compact/op_configure/op_bin_bit.rs

```diff
@@ -0,0 +1,264 @@
+        let rhs = common.alloc_u64();
+        let res = common.alloc_u64();
+        let vtype = common.alloc_common_range_value();
+
+        let is_and = common.alloc_bit_value();
+        let is_or = common.alloc_bit_value();
+        let is_xor = common.alloc_bit_value();
+        let is_not = common.alloc_bit_value();
+
+        let lookup_stack_read_lhs = common.alloc_mtable_lookup();
+        let lookup_stack_read_rhs = common.alloc_mtable_lookup();
+        let lookup_stack_write = common.alloc_mtable_lookup();
+
+        Box::new(BinBitConfig {
+            lhs,
+            rhs,
+            res,
+            vtype,
+
+            is_and,
+            is_or,
+            is_xor,
+            is_not,
+
+            lookup_stack_read_lhs,
+            lookup_stack_read_rhs,
+            lookup_stack_write,
+        })
+
+    }
+}
+
+
+impl<F: FieldExt> EventTableOpcodeConfig<F> for BinBitConfig {
+    fn opcode(&self, meta: &mut VirtualCells<'_, F>) -> Expression<F> {
+        constant!(bn_to_field(
+            &(BigUint::from(OpcodeClass::BinBit as u64) << OPCODE_CLASS_SHIFT)
+        )) + self.is_and.expr(meta)
+            * constant!(bn_to_field(
+                &(BigUint::from(BitOp::And as u64) << OPCODE_ARG0_SHIFT)
+            ))
+            + self.is_or.expr(meta)
+                * constant!(bn_to_field(
+                    &(BigUint::from(BitOp::Or as u64) << OPCODE_ARG0_SHIFT)
+                ))
+            + self.is_xor.expr(meta)
+                * constant!(bn_to_field(
+                    &(BigUint::from(BitOp::Xor as u64) << OPCODE_ARG0_SHIFT)
+                ))
+            + self.is_not.expr(meta)
+                * constant!(bn_to_field(
+                    &(BigUint::from(BitOp::Not as u64) << OPCODE_ARG0_SHIFT)
+                ))
+            + self.vtype.expr(meta)
+                * constant!(bn_to_field(&(BigUint::from(1u64) << OPCODE_ARG1_SHIFT)))
+    }
+
+    fn assign(
+        &self,
+        ctx: &mut Context<'_, F>,
+        step_info: &StepStatus,
+        entry: &EventTableEntry,
+    ) -> Result<(), Error> {
+        let (class, vtype, left, right, value ) =
+            match entry.step_info {
+                StepInfo::I32BinBitOp {
+                    class,
+                    left,
+                    right,
+                    value,
+                } => {
+                    let vtype = VarType::I32;
+                    let left = left as u32 as u64;
+                    let right = right as u32 as u64;
+                    let value = value as u32 as u64;
+                    (class, vtype, left, right, value)
+                },
+                StepInfo::I64BinBitOp {
+                    class,
+                    left,
+                    right,
+                    value,
+                } => {
+                    let vtype = VarType::I64;
+                    let left = left as u32 as u64;
+                    let right = right as u32 as u64;
```

Comment:
i64 instructions do not require type casting

Reply:
that's right, thanks! I removed it before but somehow I stashed the change

Reply:
oh, ic, it was my old commit